app/views/partials/_navbar.html.erb

```diff
@@ -1,13 +1,23 @@
-<!-- add logic to filter if the viewer is a user or a guest -->
 <!-- add url and method to line 5 once we have a path for search functionality -->
 <!-- add path and method to line 10 once we have browse functionality -->
-<section class="guest-nav">
+<section class='all-user-nav'
   <%= form_with do %>
     <%= label_tag(:q, "Search for Music") %>
     <%= text_field_tag(:q) %>
     <%= submit_tag("Search") %>
   <% end %>
   <%= button_to "Browse Music"%>
-  <%= button_to "Register", "/auth/google_oauth2" %>
-  <%= button_to "Log In", "/auth/google_oauth2" %>
 </section>
+<% if @current_user %>
```

Comment:
Let's use a helper method in the ApplicationController which we can call at any point of our app.

Reply:
can you explain more? I believe this @current_user method is from Eric's implementation of current_user that will be in the application controller, if I'm not mistaken.

Reply:
Yep current user is up and running in application controller as a neat helper method.